src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -5855,5 +5896,20 @@ private void EmitTimeoutCheckIfNeeded()
                 Call(s_checkTimeoutMethod);
             }
         }
+
+        /// <summary>
+        /// Adds an entry in <see cref="CompiledRegexRunner._indexOfAnyValues"/> for the given <paramref name="chars"/> and emits a load of that initialized value.
+        /// </summary>
+        private void LoadIndexOfAnyValues(char[] chars)
+        {
+            List<IndexOfAnyValues<char>> list = _indexOfAnyValues ??= new();
+            int index = list.Count;
+            list.Add(IndexOfAnyValues.Create(chars));
```

Comment:
This is fine for now.  We might want to look at deduplicating (e.g. with a `Dictionary<string, (int, IndexOfAnyValues<char>)>` instead of a `List<IndexOfAnyValues<char>>`) when the same character class is used multiple times in the pattern, as that's relatively common.